File: 檔案繳交/APP_10846043/ConnectionToSql_JAVA/app/src/main/java/com/example/connectiontosql_java/MainActivity.java
package com.example.connectiontosql_java;

import androidx.appcompat.app.AppCompatActivity;

import android.graphics.Bitmap;
import android.graphics.BitmapFactory;
import android.os.Bundle;
import android.util.Base64;
import android.util.Log;
import android.widget.ImageView;
import android.widget.TextView;
import android.widget.Toast;

import java.sql.Blob;
import java.sql.Connection;
import java.sql.ResultSet;
import java.sql.Statement;
import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    TextView textview;
    ImageView imageView;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        textview = findViewById(R.id.textView);
        imageView = findViewById(R.id.imageView);

        List<String> names = new ArrayList<String>();
        Connection connection;
        try {
            ConSQL c = new ConSQL();
            connection = c.conclass();
            String sqlstatement = "SELECT store_name FROM store_info";
            Statement smt = connection.createStatement();
            ResultSet set = smt.executeQuery(sqlstatement);
            while (set.next()){
                //names.add(set.getString(1));
            }
            textview.setText(names.toString());

            connection.close();
        }catch (Exception e){
            Log.d("SqlCon1",e.toString());
        }

        try {
            ConSQL c = new ConSQL();
            connection = c.conclass();
            String sqlstatement = "SELECT image FROM store_image where store_info_id = 155";
            Statement smt = connection.createStatement();
            ResultSet set = smt.executeQuery(sqlstatement);
            while (set.next()){
                //names.add(set.getString(1));
                String image = set.getString(1);

                byte[] bytes = Base64.decode(image,Base64.NO_CLOSE);
                Bitmap bitmap = BitmapFactory.decodeByteArray(bytes,0,bytes.length);


                imageView.setImageBitmap(bitmap);
                Toast.makeText(this,"Finish",Toast.LENGTH_LONG).show();
            }
            textview.setText(names.toString());

            connection.close();
        }catch (Exception e){
            Log.d("SqlCon1",e.toString());
        }

    }
}

File: 檔案繳交/APP_10846043/ConnectionToSQL/app/src/main/java/com/example/connectiontosql/MainActivity.kt
package com.example.connectiontosql

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Toast
import kotlinx.android.synthetic.main.activity_main.*
import java.sql.*

class MainActivity : AppCompatActivity() {
    lateinit var connection : Connection
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button.setOnClickListener {
            var list = mutableListOf<String>()
            try {
                val c : ConSQL = ConSQL()
                connection = c.conclass()
                var sqlstatement : String = "SELECT store_name FROM store_info"
                var smt : Statement = connection.createStatement()
                var set : ResultSet = smt.executeQuery(sqlstatement)
                while (set.next()){
                    textView.text = set.getString(1)
                    list+=set.getString(1)
                }
                connection.close()
            }catch (e : Exception){
                textView.text = e.toString()
            }

            Toast.makeText(this,list.toString(),Toast.LENGTH_LONG).show()
        }

        button2.setOnClickListener {
            try {
                val c : ConSQL = ConSQL()
                connection = c.conclass()
                var sqlstatement : String = "Insert Into restaurant_list (id, 餐廳名稱, 餐廳評價, 餐廳地址, 餐廳連結)Values (9999, 'A', 5.0, 'AA', 'AAA')"
                var smt : Statement = connection.createStatement()
                var set : Int = smt.executeUpdate(sqlstatement)
                textView.text = set.toString()

                connection.close()
            }catch (e : Exception){
                textView.text = e.toString()
            }
        }

        button3.setOnClickListener {
            try {
                val c : ConSQL = ConSQL()
                connection = c.conclass()
                var sqlstatement : String = "UPDATE restaurant_list SET 餐廳名稱= 'B' WHERE id = 9999;"
                var smt : Statement = connection.createStatement()
                var set : Int = smt.executeUpdate(sqlstatement)
                textView.text = set.toString()

                connection.close()
            }catch (e : Exception){
                textView.text = e.toString()
            }
        }

        button4.setOnClickListener {
            try {
                val c : ConSQL = ConSQL()
                connection = c.conclass()
                var sqlstatement : String = "Delete from restaurant_list"
                var smt : Statement = connection.createStatement()
                var set : Int = smt.executeUpdate(sqlstatement)
                textView.text = set.toString()

                connection.close()
            }catch (e : Exception){
                textView.text = e.toString()
            }
        }
    }
}